fix PublishEvent: Add ClientInfo

clmno opened the pull request:
In `centrifuge-go` a publish event (Publication struct) has 3 values - `Offset`, `Data` and `Info` ([view implementation](https://github.com/centrifugal/centrifuge-go/blob/595eded263578dd44cb6f2b6a8d3a08ac34b8f4a/protocol.go#L18)). But the java client only has 2 - offset and data.
This commit adds `info` to the `PublishEvent` class and sets it in the Client class.

FZambia commented on centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java:
```diff
@@ -829,10 +829,16 @@ private void handleAsyncReply(Protocol.Reply reply) {
                 if (sub != null) {
                     PublishEvent event = new PublishEvent();
                     event.setData(pub.getData().toByteArray());
+                    event.setInfo(info);
```
Should not be set if it's not set in incoming Publication - i.e. should be null, not an empty ClientInfo object as done here.

FZambia commented on centrifuge/src/main/java/io/github/centrifugal/centrifuge/PublishEvent.java:
```diff
@@ -11,6 +11,17 @@ void setData(byte[] data) {
 
     private byte[] data;
 
+    public ClientInfo getInfo() {
```
Could you please also add this to `ServerPublishEvent` and set it accordingly?

FZambia reviewed (changes requested):
Thanks! Added some review comments

clmno commented:
@FZambia I've made the changes you've requested.
1. ClientInfo set to `null` if the protocol.ClientInfo == null
2. Added ClientInfo for `PublishEvent`
2. Added the same for `ServerPublishEvent`

Please review and let me know

clmno reviewed (commented):
I've made the necessary changes

FZambia commented:
@clmno great, thanks!
